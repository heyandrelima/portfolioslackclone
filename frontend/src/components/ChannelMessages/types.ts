import { Message } from '../../types/api';

export type Props = {
    messages: Message[];
}
