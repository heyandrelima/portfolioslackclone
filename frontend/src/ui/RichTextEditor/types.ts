export type ContainerProps = {
    menuHeight: number;
}