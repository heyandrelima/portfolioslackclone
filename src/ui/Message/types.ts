import { Message } from '../../types/api';

export type Props = {
    message: Message
}
