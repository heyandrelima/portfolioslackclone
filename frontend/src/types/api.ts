export type Author = {
    name: string;
    avatar: string;
}

export type Message = {
    createdAt: Date;
    id: string;
    author: Author;
    body: string;
}

export type Channel = {
    id: string;
    label: string;
    description?: string;
    messages?: Array<Message>;
}
