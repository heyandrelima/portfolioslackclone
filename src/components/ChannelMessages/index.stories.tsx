import ThemeContext from '../../context/Theme';
import { defaultTheme } from '../../themes';
import { Author, Message } from '../../types/api';
import ChannelMessages from './index';

const Story = {
    title: 'Components/ChannelMessages',
    component: ChannelMessages,
};

export const Default = () => {
    const author: Author = {
        name: 'Admin',
        avatar: 'http://andrelima.space/static/media/profile.285b1009.jpeg',
    };
    const messages: Message[] = [
        {
            id: 'abc',
            author,
            body: 'Hello everyone!',
            createdAt: new Date('2021-01-01 16:00:00'),
        },
        {
            id: 'def',
            author,
            body: 'Second message',
            createdAt: new Date('2021-01-01 16:01:00'),
        },
        {
            id: 'ab12',
            author,
            body: 'Last message',
            createdAt: new Date('2021-01-01 16:05:00'),
        },
    ];
    return (
        <ThemeContext theme={defaultTheme}>
            <ChannelMessages messages={messages} />
        </ThemeContext>
    );
};

export default Story;
