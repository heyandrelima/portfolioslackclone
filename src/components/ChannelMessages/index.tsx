import React from 'react';
import { Message as MessageType } from '../../types/api';
import Message from '../../ui/Message';
import { Props } from './types';
import { Container } from './styles';

const ChannelMessages: React.FC<Props> = (props: Props) => (
    <Container data-testid='channel-messages'>
        {props.messages.map((message: MessageType, index: number) => (
            <Message message={message} key={index} />
        ))}
    </Container>
);

export default ChannelMessages;
