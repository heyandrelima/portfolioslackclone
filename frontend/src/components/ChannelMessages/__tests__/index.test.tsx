import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ChannelMessages from '../';
import { Author, Message } from '../../../types/api';

describe('components/ChannelMessages', () => {
    it('shoud render correctly', () => {
        const author: Author = {
            name: 'Admin',
            avatar: '/avatar.jpg',
        };
        const messages: Message[] = [
            {
                id: 'abc',
                author,
                body: 'Hello everyone!',
                createdAt: new Date('2021-01-01 16:00:00'),
            },
            {
                id: 'def',
                author,
                body: 'Second message',
                createdAt: new Date('2021-01-01 16:01:00'),
            },
            {
                id: 'ab12',
                author,
                body: 'Last message',
                createdAt: new Date('2021-01-01 16:05:00'),
            },
        ];
        const component = render(<ChannelMessages messages={messages} />).queryByTestId('channel-messages');
        expect(component).toMatchSnapshot();
    });
});
