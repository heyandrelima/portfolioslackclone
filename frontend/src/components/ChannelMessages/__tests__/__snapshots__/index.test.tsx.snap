// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ChannelMessages shoud render correctly 1`] = `
<div
  class="sc-pFZIQ dJJXEx"
  data-testid="channel-messages"
>
  <div
    class="sc-bdfBwQ ktSfRd"
  >
    <div
      class="sc-gsTCUz hvBwxn"
    >
      <div
        class="sc-hKgILt ezYlzq"
      >
        <img
          alt="avatar"
          class="sc-eCssSg laFpUa"
          src="/avatar.jpg"
        />
      </div>
    </div>
    <div
      class="sc-dlfnbm cMmjyT"
    >
      <div
        class="sc-jSgupP gbYrXe"
      >
        <p
          class="sc-gKsewC gsTsyj"
          data-testid="author"
        >
          Admin
        </p>
        <p
          class="sc-iBPRYJ cXPabg"
          data-testid="time"
        >
          04:00 PM
        </p>
      </div>
      <p
        class="sc-fubCfw gqInqL"
        data-testid="message"
      >
        Hello everyone!
      </p>
    </div>
  </div>
  <div
    class="sc-bdfBwQ ktSfRd"
  >
    <div
      class="sc-gsTCUz hvBwxn"
    >
      <div
        class="sc-hKgILt ezYlzq"
      >
        <img
          alt="avatar"
          class="sc-eCssSg laFpUa"
          src="/avatar.jpg"
        />
      </div>
    </div>
    <div
      class="sc-dlfnbm cMmjyT"
    >
      <div
        class="sc-jSgupP gbYrXe"
      >
        <p
          class="sc-gKsewC gsTsyj"
          data-testid="author"
        >
          Admin
        </p>
        <p
          class="sc-iBPRYJ cXPabg"
          data-testid="time"
        >
          04:01 PM
        </p>
      </div>
      <p
        class="sc-fubCfw gqInqL"
        data-testid="message"
      >
        Second message
      </p>
    </div>
  </div>
  <div
    class="sc-bdfBwQ ktSfRd"
  >
    <div
      class="sc-gsTCUz hvBwxn"
    >
      <div
        class="sc-hKgILt ezYlzq"
      >
        <img
          alt="avatar"
          class="sc-eCssSg laFpUa"
          src="/avatar.jpg"
        />
      </div>
    </div>
    <div
      class="sc-dlfnbm cMmjyT"
    >
      <div
        class="sc-jSgupP gbYrXe"
      >
        <p
          class="sc-gKsewC gsTsyj"
          data-testid="author"
        >
          Admin
        </p>
        <p
          class="sc-iBPRYJ cXPabg"
          data-testid="time"
        >
          04:05 PM
        </p>
      </div>
      <p
        class="sc-fubCfw gqInqL"
        data-testid="message"
      >
        Last message
      </p>
    </div>
  </div>
</div>
`;
